program/freedom-metal-program.wake

```diff
@@ -1,4 +1,28 @@
 # FIXME
+def addMetalMemory type memSubnode inDTSContents =
+  def subnodeOpt =
+    match "({memSubnode}@[0-9]+)".stringToRegExp
+      Pass regex =
+        def nodeRegex = regExpCat (`^\t\t`, `(?:L[0-9]+:\s)?`, regex, `\s{$`, Nil)
+        inDTSContents
+        | map nodeRegex.extract
+        | flatten
+        | head
+        | omap ("/soc/{_}")
+      Fail _ = None
+
+  match subnodeOpt
+    Some subnode =
+      def asdf =
+        inDTSContents
+        | mapFlat (\a (
+          if matches `^\tchosen\s{$` a
+          then ("\tchosen \{", "\t\tmetal,{type} = <&\{{subnode}\}>;", Nil)
```

Comment:
Perhaps you could consider changing the `type` variable to include the `metal,` prefix, so that this is an entire Device Tree property name? This would make it clearer how the specific type name is combined with the `metal,` prefix at the call site.